Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -103,12 +121,73 @@ public struct DiagnosticsFormatter {
         }
 
         for diag in diags.dropLast(1) {
-          annotatedSource.append("\(preMessage)├─ \(diag.message)\n")
+          annotatedSource.append("\(preMessage)├─ \(colorizeIfRequested(diag.diagMessage))\n")
         }
-        annotatedSource.append("\(preMessage)╰─ \(diags.last!.message)\n")
-
+        annotatedSource.append("\(preMessage)╰─ \(colorizeIfRequested(diags.last!.diagMessage))\n")
       }
     }
     return annotatedSource
   }
+
+  /// Annotates the given ``DiagnosticMessage`` with an appropriate ANSI color code (if the value of the `colorize`
+  /// property is `true`) and returns the result as a printable string.
+  private func colorizeIfRequested(_ message: DiagnosticMessage) -> String {
+    guard colorize else {
+      return message.message
+    }
+
+    switch message.severity {
+    case .error:
+      let annotation = ANSIAnnotation(color: .red, trait: .bold)
+      return annotation.applied(to: message.message)
+    case .warning:
+      let annotation = ANSIAnnotation(color: .yellow)
+      return annotation.applied(to: message.message)
+    case .note:
+      return message.message
+    }
+  }
+}
+
+struct ANSIAnnotation {
+  enum Color: UInt8 {
+    case normal = 0
+    case black = 30
+    case red = 31
+    case green = 32
+    case yellow = 33
+    case blue = 34
+    case magenta = 35
+    case cyan = 36
+    case white = 37
+  }
+
+  enum Trait: UInt8 {
+    case normal = 0
+    case bold = 1
+    case underline = 4
+  }
+
+  var color: Color
+  var trait: Trait
+
+  /// The textual representation of the annotation.
+  var code: String {
+    "\u{001B}[\(trait.rawValue);\(color.rawValue)m"
+  }
+
+  init(color: Color, trait: Trait = .normal) {
+    self.color = color
+    self.trait = trait
+  }
+
+  func applied(to message: String) -> String {
+    // Resetting after the message ensures that we don't color unintended lines in the output
+    return "\(code)\(message)\(ANSIAnnotation.reset.code)"
+  }
+
+  /// The "reset" ANSI code used to unset any previously added annotation.
+  static var reset: ANSIAnnotation {
```

Comment:
To be consistent with `normal` in `Color` and `Trait`, should this also be named `normal`?

Reply:
Sure, that sounds good to me 👍🏻